File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/general/GeneralSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.general;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.github.alexandrecarlton.idea.settings.layout.editor.general.auto_import.AutoImportSettings;
import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableGeneralSettings.class)
public interface GeneralSettings {
  Optional<AutoImportSettings> autoImport();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/editor/general/GeneralSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.editor.general

import com.fasterxml.jackson.annotation.JsonProperty
import com.github.alexandrecarlton.idea.settings.layout.editor.general.auto_import.AutoImportSettings

data class GeneralSettings(
    @JsonProperty("Auto Import")
    val autoImport: AutoImportSettings? = null
)
